src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1619,32 +1618,35 @@ private RefreshItem getRefreshToken(final AuthenticationRequest request) {
         final String methodName = ":getRefreshToken";
         RefreshItem refreshItem = null;
         if (mTokenCacheStore != null) {
-            boolean multiResource = false;
-            // target refreshToken for this resource first. CacheKey will
-            // include the resourceId in the cachekey
-            Logger.v(TAG, "Looking for regular refresh token");
             String userId = request.getUserId();
             if (StringExtensions.IsNullOrBlank(userId)) {
                 // acquireTokenSilent expects userid field from UserInfo
                 userId = request.getLoginHint();
             }
-            String keyUsed = CacheKey.createCacheKey(request, userId);
+
+            // Target multiResource item in cache first because it's the most recent one.
+            // Cache key will not include resourceId.
+            Logger.v(TAG, "Looking for Multi Resource Refresh token");
+            String keyUsed  = CacheKey.createMultiResourceRefreshTokenKey(request, userId);
             TokenCacheItem item = mTokenCacheStore.getItem(keyUsed);
+
             if (item == null || StringExtensions.IsNullOrBlank(item.getRefreshToken())) {
-                // if not present, check multiResource item in cache. Cache key
-                // will not include resourceId in the cache key string.
-                Logger.v(TAG, "Looking for Multi Resource Refresh token");
-                keyUsed = CacheKey.createMultiResourceRefreshTokenKey(request, userId);
+                // if not present, check refreshToken for specified resource.
+                // CacheKey will include the resourceId
+                Logger.v(TAG, "Looking for regular refresh token");
+                keyUsed = CacheKey.createCacheKey(request, userId);
                 item = mTokenCacheStore.getItem(keyUsed);
-                multiResource = true;
             }
             
-            // If still cannot find an item, try with family clientid
-            // Disable this feature for ADFS authority
             if (item == null || StringExtensions.IsNullOrBlank(item.getRefreshToken())) {
+                // If still cannot find an item, try with family clientid
+                // Disable this feature for ADFS authority
                 if (!UrlExtensions.isADFSAuthority(StringExtensions.getUrl(mAuthority))) {
                     Logger.v(TAG + methodName, "No refresh token found, trying to find family client id item.");
-                    item = findFamilyItemForUser(request.getUserIdentifierType(), userId);
+                    item = findFamilyItemForUser(request.getUserIdentifierType(), userId, request.getClientId());
```

Comment:
@imakhalova  We are going to temporarily disable Family of Client ID support.  The server is going to change its implementation to give us a single FoCI RT that can be used across the entire set of client ids so that we don't have to use buggy heuristics to find the most proofed up FoCI RT.  @weijjia Will send out a PR for that relatively soon.  Unless you want to disable it in this PR.